scipy/stats/tests/test_stats.py

```diff
@@ -2178,15 +2178,21 @@ def test_negative_axes_gh_15375(self, axis):
 
     def test_strings(self):
         data1 = ['rain', 'showers', 'showers']
-        vals = stats.mode(data1)
+        with pytest.warns(DeprecationWarning,
+                          match=r"Support for non-numeric arrays has been "
+                                r"deprecated"):
```

Comment:
NBD since this is going to be removed in two versions anyway, but consider:
```suggestion
            message = "Support for non-numeric arrays has been deprecated"
            with pytest.warns(DeprecationWarning, match=message):
```

Reply:
Done.